source/common/http/conn_manager_impl.cc

```diff
@@ -772,6 +772,8 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(RequestHeaderMapPtr&& he
                                connection_manager_.read_callbacks_->connection().dispatcher());
   request_headers_ = std::move(headers);
 
+  clusterInfo()->stats().upstream_rq_headers_size_.recordValue(request_headers_->byteSize());
```

Comment:
you can probably access cluster_info object via `connection_manager_.read_callbacks_->clusterInfo()` and not add extra method to ActiveStream api.

Reply:
Got `clusterinfo()` by `connection_manager_.read_callbacks_->upstreamHost()->cluster()`. Couldn't find `clusterInfo()` directly part of `read_callbacks_`.